builtin/credential/approle/validation.go

```diff
@@ -68,18 +68,25 @@ type secretIDAccessorStorageEntry struct {
 
 // verifyCIDRRoleSecretIDSubset checks if the CIDR blocks set on the secret ID
 // are a subset of CIDR blocks set on the role
-func verifyCIDRRoleSecretIDSubset(secretIDCIDRs []string, roleBoundCIDRList []string) error {
-	if len(secretIDCIDRs) != 0 {
-		// If there are no CIDR blocks on the role, then the subset
-		// requirement would be satisfied
-		if len(roleBoundCIDRList) != 0 {
-			subset, err := cidrutil.SubsetBlocks(roleBoundCIDRList, secretIDCIDRs)
-			if !subset || err != nil {
-				return errwrap.Wrapf(fmt.Sprintf("failed to verify subset relationship between CIDR blocks on the role %q and CIDR blocks on the secret ID %q: {{err}}", roleBoundCIDRList, secretIDCIDRs), err)
-			}
-		}
+func verifyCIDRRoleSecretIDSubset(secretIDCIDRs []string, role *roleStorageEntry) error {
+	if len(secretIDCIDRs) == 0 {
+		return nil
 	}
 
+	// If there are no CIDR blocks on the role, then the subset
+	// requirement would be satisfied
+	if len(role.SecretIDBoundCIDRs) != 0 {
+		subset, err := cidrutil.SubsetBlocks(role.SecretIDBoundCIDRs, secretIDCIDRs)
+		if !subset || err != nil {
+			return errwrap.Wrapf(fmt.Sprintf("failed to verify subset relationship between CIDR blocks on the role %q and CIDR blocks on the secret ID %q: {{err}}", role.SecretIDBoundCIDRs, secretIDCIDRs), err)
+		}
+	}
+	if len(role.TokenBoundCIDRs) != 0 {
```

Comment:
I wonder if we really need this check. We are only ever going to pass in only the secret ID CIDRs to this function for subset checking. Consider that secret IDs are generated by a orchestration module sitting under a different CIDR than the nodes that will be _using_ the tokens belonging to a disjoint CIDR block. Then essentially what we are asserting here is that both during login time and when updating secret IDs, the secret ID CIDRs to be a subset of TokenBoundCIDRs.

I am thinking that we are better off leaving the TokenBoundCIDRs out of this function.

Reply:
Ah! Thank you!